src/TempleLineOfCreditV2.sol

```diff
@@ -0,0 +1,368 @@
+
+    /// @notice contract to get TPI price
+    IOudRedeemer public oudRedeemer;
+
+    constructor(
+        address _templeToken,
+        TokenPrice _templePrice,
+
+        address _daiToken,
+        address _daiInterestRateModel,
+        TokenPrice _daiPrice,
+        uint256 _daiMinCollateralizationRatio,
+
+        address _oudToken,
+        address _oudInterestRateModel,
+        TokenPrice _oudPrice,
+        uint256 _oudMinCollateralizationRatio,
+
+        address _oudRedeemer
+
+    ) {
+
+        templeToken = IERC20(_templeToken);
+        templePrice = _templePrice;
+
+        dai.tokenAddress =_daiToken;
+        dai.interestRateModel = _daiInterestRateModel;
+        dai.tokenPrice = _daiPrice;
+        dai.minCollateralizationRatio = _daiMinCollateralizationRatio; 
+
+        oud.tokenAddress = _oudToken;
+        oud.interestRateModel = _oudInterestRateModel;
+        oud.tokenPrice = _oudPrice;
+        oud.minCollateralizationRatio = _oudMinCollateralizationRatio;
+
+        oudRedeemer = IOudRedeemer(_oudRedeemer);
+    }
+
+    function addOperator(address _address) external override onlyOwner {
+        _addOperator(_address);
+    }
+
+    function removeOperator(address _address) external override onlyOwner {
+        _removeOperator(_address);
+    }
+
+    /**
+     * @notice Set the borrow interest rate for debt token
+     * @param debtToken address of the debt token
+     * @param interestRateModel the interest rate model to use
+     */
+    function setInterestRateModel(address debtToken, address interestRateModel) external onlyOperators{
+
+        if (debtToken == dai.tokenAddress) {
+            _accrueInterest(dai);
+            dai.interestRateModel = interestRateModel;
+        } else if (debtToken == oud.tokenAddress) {
+            _accrueInterest(oud);
+            oud.interestRateModel = interestRateModel;
+        } else {
+            revert Unsupported(debtToken);
+        }
+    }
+
+
+    function _accrueInterest(ReserveToken storage reserve) internal {
+
+        uint256 totalBorrow = reserve.totalBorrow;
+        uint256 totalReserve = reserve.totalReserve;
+        uint256 totalShares = reserve.totalShares;
+
+        // If no borrow just reset interest updated at.
+        if (totalShares == 0) {
+
+            reserve.interestRateLastUpdatedAt = block.timestamp;
+
+        } else {
+
+            uint256 timeElapsed = block.timestamp - reserve.interestRateLastUpdatedAt;
+            uint256 newInterestRate =  IInterestRateModel(reserve.interestRateModel).getBorrowRate(totalBorrow, totalReserve);
+
+            uint256 interestEarned = totalBorrow.continuouslyCompounded(timeElapsed, newInterestRate) - totalBorrow;
+
+            totalBorrow += interestEarned;
+            totalReserve += interestEarned; // Reserve should also increase to properly account for utilization rate
+
+            reserve.totalBorrow = totalBorrow;
+            reserve.totalReserve = totalReserve;
```

Comment:
Hmm the contract doesn't actually have the token in it's balance to allower users to borrow at this point though.

Reply:
but we still a concept of total reserve to account for utilization. Total reserve can mean here the amount that tlc as put inside the treasureReserveVault.